.github/workflows/ci.yml

```diff
@@ -221,47 +212,51 @@ jobs:
 
     strategy:
       matrix:
-        compiler: ["gcc-13", "clang-18"]
+        os:
+          - "ubuntu-latest"
+        compiler:
+          - "gcc-11"
+          - "gcc-13"
+          - "clang-18"
 
-    runs-on: ubuntu22_4c_16gb
+        include:
+          - os: "windows-2022"
+            compiler: "msvc"
+
+    runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v4
         with:
           lfs: true
 
       - name: General Build Dependencies
+        if: contains(matrix.os, 'ubuntu')
         run: |
-          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
-          sudo apt-get update -q
           sudo apt-get install ninja-build -y -q
 
-      - name: Install GCC
-        if: matrix.compiler == 'gcc-13'
-        run: >-
-          sudo apt-get install ninja-build gcc-13 g++-13 g++-13-multilib -y -q
+      - uses: ./.github/actions/install-gcc
+        if: ${{ contains(matrix.compiler, 'gcc') && contains(matrix.os, 'ubuntu') }}
+        with:
+          version: ${{ matrix.compiler }}
 
-      - name: Install LLVM and Clang
-        if: matrix.compiler == 'clang-18'
-        uses: ./.github/actions/install-clang
+      - uses: ./.github/actions/install-clang
+        if: ${{ contains(matrix.compiler, 'clang') && contains(matrix.os, 'ubuntu') }}
```

Comment:
do we need the ubuntu check here?

Reply:
The llvm-script only works on systems using apt.
I can remove it since our matrix is either windows or ubuntu?